src/index.ts

```diff
@@ -955,12 +958,26 @@ export class BufferedChangeset implements IChangeset {
           const baseContent = this.safeGet(content, baseKey);
           const subContent = this.getDeep(baseContent, remaining.join('.'));
           const subChanges = getSubObject(changes, key);
-          // give back and object that can further retrieve changes and/or content
+          // give back an object that can further retrieve changes and/or content
           const tree = new ObjectTreeNode(subChanges, subContent, this.getDeep, this.isObject);
           return tree.proxy;
         } else if (typeof result !== 'undefined') {
           return result;
         }
+        const baseContent = this.safeGet(content, baseKey);
+
+        if (Array.isArray(baseContent)) {
+          const subChanges = getSubObject(changes, key);
+
+          if (!subChanges) {
+            return this.getDeep(baseContent, remaining.join('.'));
+          }
+
+          // give back an object that can further retrieve changes and/or content
+          const tree = new ObjectTreeNode(subChanges, baseContent, this.getDeep, this.isObject);
+
+          return tree;
```

Comment:
For the array case, do we need to return an ObjectTreeNode?  Just a little background just in case there is some intersting information for this discussion...

The main reason for the `ObjectTreeNode` is to ensure we don't lose sibling keys.  So an object with both `changes` and unchanged content are able to be retrieved at any arbitrary level of the object. 

```
{
  changeValue: ...,
  original: ... 
}
```

Now for the array scenario, in one case we have `changes` we want to return.  In the other, we have `content` we want to return.  However, in both cases, I imagine that the type is an array and we don't have to deal with merging sibling keys.

Reply:
> For the array case, do we need to return an ObjectTreeNode? 

yeah, cause atm, ObjectTreeNode is the only thing that knows that changes are on arrays and can convert between arrays and array objects.

_however_ this changes the behavior of `changeset.get('prop.someArray')` where it now needs `unwrap`, where it previously did not -- but this is more consistent with the deep object changes.

> . However, in both cases, I imagine that the type is an array and we don't have to deal with merging sibling keys.

🤷 
The overloaded behavior of this method is dizzying to me :(